runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java

```diff
@@ -41,14 +48,23 @@
 	tags = {"limit", "vertical"},
 	enabledByDefault = false
 )
-public class ZoomPlugin extends Plugin
+public class ZoomPlugin extends Plugin implements KeyListener
 {
 	@Inject
 	private Client client;
 
 	@Inject
 	private ZoomConfig zoomConfig;
 
+	@Inject
+	private KeyManager keyManager;
+
+	@Inject
+	private EventBus eventBus;
+
+	private boolean loggedIn;
+	private boolean hotkeyDown;
```

Comment:
You should reset hotkeyDown on focus change. Also, you can just use the Hotkey class we have instead of listener